### ISSUE TITLE
[RTL] Border issue for first column.

### ISSUE THREAD

**Issue by Siemienik** [handsontable/handsontable#8767]
### Description

When I set `dir='rtl'` the first row has still bordered on the left. It should be moved to the right because it is a border with row headers.
* [x] Additionally, the StartOverlay should have got a border on the right side (#8763) 

### Steps to reproduce

Following this [configuration](https://github.com/Siemienik/handsontable-storybooks/blob/f229aab5311a146c4a84b1892c2b0b57d291e216/vanilla/configurations.js#L5-L10) 

```js
configurations.JustATable = {
    data: Handsontable.helper.createSpreadsheetData(6, 10),
    colHeaders: true,
    rowHeaders: true,
    licenseKey: "non-commercial-and-evaluation",
};
```
I got:  (incorrect border is in the reactangle)
![image](https://user-images.githubusercontent.com/2893099/135144556-5e9cd582-44c8-4700-a97f-637625303dec.png)

**Comment by Siemienik** [comment]
done in #8889

**Comment by AMBudnik** [comment]
Published in Handsontable v11.1.0

**Comment by aninde** [comment]
Solved in the v12.0.0 https://jsfiddle.net/t69meyL0/

**Comment by AMBudnik** [comment]
Closed with https://github.com/handsontable/handsontable/issues/8760 and developed and done in Handsontable v12.

For more information please check out our Release Notes - https://handsontable.com/docs/release-notes/#_12-0-0
